useIsShake.js

```diff
@@ -1,59 +1,44 @@
 import React, { useState, useEffect } from 'react';
 import { Accelerometer } from 'expo-sensors';
-import { QUOTES_IN_DURATION, QUOTES_IN_DELAY, QUOTES_OUT_DURATION, SVG_LOAD_DURATION, SVG_OUT_DURATION, SVG_IN_DURATION, SVG_IN_DELAY, TRANSITION_TIME_2 } from './constants';
 
 export function useIsShake() {
   const [{ x }, setData] = useState({
     x: 0,
   });
   const [subscription, setSubscription] = useState(null);
-  const [toggle, setToggle] = useState(false);
+  const [isShakeTriggered, setIsShakeTriggered] = useState(false);
+  const [isToggleReady, setIsToggleReady] = useState(false);
 
   let polls = [null, null];
   let diffs = [];
-  let shakeCount = 0;
-  let isToggleReady;
-  let hasToggledBefore;
 
   function isShake({ x: newX }) {
-    if (isToggleReady === true) {
-      if (diffs.length === 2) {
-        shakeCount++;
-        setToggle(true);
-        isToggleReady = false;
+    // console.log(newX);
 
-        if (hasToggledBefore === undefined) hasToggledBefore = false;
-        else if (hasToggledBefore === false) hasToggledBefore = true;
+    if (diffs.length === 2) {
+      setIsShakeTriggered(true);
+      setIsToggleReady(false);
```

Comment:
If a shake has been registered, indicated by two values being present in diffs [], isShakeTriggered set true in order to fire animation and isToggleReady set false to remove Accelerometer listener.